jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/scopes/SimpleContainerScope.java

```diff
@@ -166,4 +133,44 @@ public void removeSessionListener(WebSocketSessionListener listener)
     {
         return sessionListeners;
     }
+
+    @Override
+    protected void doStart() throws Exception
```

Comment:
I don't think it's right to move the initialization code that before was in the constructor to `doStart()`.
For example, `WebSocketClient` constructor calls `HttpClientProvider.get(scope)` passing in a scope that is not yet started, which get passed to `DefaultHttpClientProvider.newHttpClient(scope)` which uses the scope before it's started.
I think we need a better review of the lifecycle of the scope to move things to a different place.